### DISCUSSION TITLE
Force the bitplatform template to always use wasm

### DISCUSSION THREAD

**Discussion by Kyemate** [bitfoundation/bitplatform#9093] [Q&A] [answered]
Hi, 
I'm trying to integrate the bitplatform template into my project. I want it to always run in the wasm mode. I enabled pwa but whatever I do it always seems to want to run on the websocket connection.

Edit when trying to force the rendermode in app.razor I get this error
![image](https://github.com/user-attachments/assets/b8dacef4-0fd6-49f4-b4db-13bb17adc0be)

**Comment by yasmoradi** [comment]
[todo.bitplatform.dev](https://todo.bitplatform.dev), [adminpanel.bitplatform.dev](https://adminpanel.bitplatform.dev), [blazorui demo](https://blazorui.bitplatform.dev) and [bitplatform.dev](https://bitplatform.dev) itself are all using blazor webassembly only in production, so that's possible to have such a thing
Refer to the docs to figure out how to change blazor mode in [Blazor mode selection](https://bitplatform.dev/templates/run-project)

**Reply by Kyemate** [reply]
> [todo.bitplatform.dev](https://todo.bitplatform.dev), [adminpanel.bitplatform.dev](https://adminpanel.bitplatform.dev), [blazorui demo](https://blazorui.bitplatform.dev) and [bitplatform.dev](https://bitplatform.dev) itself are all using blazor webassembly only in production, so that's possible to have such a thing Refer to the docs to figure out how to change blazor mode in [Blazor mode selection](https://bitplatform.dev/templates/run-project)

Thanks for the quick answer. I tried to download a completely new template now and only changing the blazor mode selection from the docs. But I still get the same IOptionsSnapshot<ClientAppsettings> error

**Reply by yasmoradi** [reply]
Which version are you using? Could you please do these steps:
1- Create project and push it into GitHub repo without doing any change
2- Change the blazor mode and push it in separate commit
3- Live repo address here

**Reply by Kyemate** [reply]
> Which version are you using? Could you please do these steps: 1- Create project and push it into GitHub repo without doing any change 2- Change the blazor mode and push it in separate commit 3- Live repo address here

Here is the repo https://github.com/Kyemate/BitplatformWasmMode

**Reply by yasmoradi** [reply]
I'm testing your repo
But why you're changing appsettings.Development.json? Blazor server has better developer experience and even during development, you can test blazor webassembly by running client.web project at anytime!
For the production, we do recommend blazor auto or webassembly though

**Reply by Kyemate** [reply]
> I'm testing your repo But why you're changing appsettings.Development.json? Blazor server has better developer experience and even during development, you can test blazor webassembly by running client.web project at anytime! For the production, we do recommend blazor auto or webassembly though

I'm changing it for testing proposes. I'm porting over an app built on the old hosted wasm template to your template. To be sure everything works I want to run it in wasm mode. When I get it fully ported over I will switch back to Blazor server for development. In prod it will run fully in wasm mode with pwa and eventually maui.

**Reply by Kyemate** [reply]
I tried changing this line 
`services.AddTransient(sp => sp.GetRequiredService<IOptionsSnapshot<ClientAppSettings>>().Value);`

to this
`services.AddTransient(sp => sp.GetRequiredService<IOptions<ClientAppSettings>>().Value);`

in Client.Web.Program.Services.cs and now it seems to be working.

The weird thing is this line seems to work thou.
`services.AddTransient(sp => sp.GetRequiredService<IOptionsSnapshot<SharedAppSettings>>().Value);`

**Reply by yasmoradi** [reply, marked as answer]
We're going to publish stable release within the next 24 hours, I'd recommend you to use that instead of pre-release
I tested with latest codes and did't saw the issue you're mentioning
and I wasn't able to run your sample code in github codespace because it requires sql server.

**Reply by Kyemate** [reply]
> We're going to publish stable release within the next 24 hours, I'd recommend you to use that instead of pre-release I tested with latest codes and did't saw the issue you're mentioning and I wasn't able to run your sample code in github codespace because it requires sql server.

Alright, thanks for the quick replies
